lib/go-tc/deliveryservices.go

```diff
@@ -58,59 +58,60 @@ type DeleteDeliveryServiceResponse struct {
-	DSCP                 int                    `json:"dscp"`
-	EdgeHeaderRewrite    string                 `json:"edgeHeaderRewrite"`
-	ExampleURLs          []string               `json:"exampleURLs"`
-	GeoLimit             int                    `json:"geoLimit"`
-	FQPacingRate         int                    `json:"fqPacingRate"`
-	GeoProvider          int                    `json:"geoProvider"`
-	GlobalMaxMBPS        int                    `json:"globalMaxMbps"`
-	GlobalMaxTPS         int                    `json:"globalMaxTps"`
-	HTTPBypassFQDN       string                 `json:"httpBypassFqdn"`
-	ID                   int                    `json:"id"`
-	InfoURL              string                 `json:"infoUrl"`
-	InitialDispersion    float32                `json:"initialDispersion"`
-	IPV6RoutingEnabled   bool                   `json:"ipv6RoutingEnabled"`
-	LastUpdated          *TimeNoMod             `json:"lastUpdated" db:"last_updated"`
-	LogsEnabled          bool                   `json:"logsEnabled"`
-	LongDesc             string                 `json:"longDesc"`
-	LongDesc1            string                 `json:"longDesc1"`
-	LongDesc2            string                 `json:"longDesc2"`
-	MatchList            []DeliveryServiceMatch `json:"matchList,omitempty"`
-	MaxDNSAnswers        int                    `json:"maxDnsAnswers"`
-	MidHeaderRewrite     string                 `json:"midHeaderRewrite"`
-	MissLat              float64                `json:"missLat"`
-	MissLong             float64                `json:"missLong"`
-	MultiSiteOrigin      bool                   `json:"multiSiteOrigin"`
-	OrgServerFQDN        string                 `json:"orgServerFqdn"`
-	ProfileDesc          string                 `json:"profileDescription"`
-	ProfileID            int                    `json:"profileId,omitempty"`
-	ProfileName          string                 `json:"profileName"`
-	Protocol             int                    `json:"protocol"`
-	QStringIgnore        int                    `json:"qstringIgnore"`
-	RangeRequestHandling int                    `json:"rangeRequestHandling"`
-	RegexRemap           string                 `json:"regexRemap"`
-	RegionalGeoBlocking  bool                   `json:"regionalGeoBlocking"`
-	RemapText            string                 `json:"remapText"`
-	RoutingName          string                 `json:"routingName"`
-	SigningAlgorithm     string                 `json:"signingAlgorithm" db:"signing_algorithm"`
-	TypeID               int                    `json:"typeId"`
-	Type                 string                 `json:"type"`
-	TRResponseHeaders    string                 `json:"trResponseHeaders"`
-	TenantID             int                    `json:"tenantId,omitempty"`
-	XMLID                string                 `json:"xmlId"`
+	Active                   bool                   `json:"active"`
+	AnonymousBlockingEnabled bool                   `json:"anonymousBlockingEnabled"`
```

Comment:
Why was this moved to DeliveryService (v1.2)? This field is new in API 1.3, right? So it should only be in `DeliveryServiceV13` returned by `/api/1.3/`, and not `/api/1.2/...`, per Semantic Versioning

Reply:
Its in the 1.2 API: https://github.com/apache/trafficcontrol/blob/master/traffic_ops/app/lib/API/Deliveryservice.pm#L104

Reply:
But it was committed on May 14, 2018. The 1.2 API was standardized long before that.

It's kind of understandable in Perl, because the Perl app doesn't have a decent way to do Semantic Versioning. But Go does, and we agreed to use Semantic Versioning on the mailing list.

>https://semver.org/
>2. MINOR version when you add functionality in a backwards-compatible manner

Is there a big reason to add this to `/api/1.2/` in violation of SemVer?

Reply:
i would say leave the Perl alone and take this out so 1.2 is not affected

https://github.com/apache/trafficcontrol/blob/master/traffic_ops/app/lib/API/Deliveryservice.pm#L104

and then just add AnonymousBlockingEnabled to the v13 DS struct. does that make sense?

Reply:
Yes, that makes sense. I'll make the change.

Reply:
We have existing 1.2 API callers that use the anonymous blocking field. Moving it to 1.3 in the go API would break 1.2 for us (and that is not backwards compatible as SemVer claims to be). 

Really it should not have been merged to 1.2 in the first place if the API was fixed back in May, but here we are now.

Reply:
Given the scenario, I will restore the field in the v12 struct. Thank you all for the feedback.

Reply:
Eh, it makes me really sad. We really, really need to stop breaking the API promise. We're going to lose new users, because our API is unreliable and unstable.

But I won't -1 this, since at least @limited is already using it, and it's in 2.2.0-RC6

:(

Reply:
Im with you Rob, we definitely need to improve on this and have stricter API versioning. It pains me that we didnt catch it sooner and make sure that it went into 1.3.